tools/bin/ecflow/processes/main/meteorological/combine.py

```diff
@@ -0,0 +1,110 @@
+#!/usr/bin/env python
+#####combine.py
+#####usage: <python> <combine.py> <configuration.cfg>
+
+###### This script runs tocel_combine.py
+
+import sys
+import pandas as pd
+import numpy as np
+import argparse
+import types
+import os
+import multiprocessing as mp
+from shutil import copyfile
+
+from tonic.io import read_config
+from monitor.io import replace, proc_subprocess
+from monitor.log import set_logger
+from monitor.share import LOG_LEVEL 
+
+######### ----------------------------------------###########
+
+
+def main():
+	# parse arguments
+	parser = argparse.ArgumentParser(description='Run VIC')
+	parser.add_argument('config_file', metavar='config_file',
+                        type=argparse.FileType('r'), nargs=1,
+                        help='configuration file')
+	args = parser.parse_args()
+	config_dict = read_config(args.config_file[0].name)
+
+	# set up logger
+	logger = set_logger(os.path.splitext(os.path.split(__file__)[-1])[0],
+                        LOG_LEVEL)
+
+	#read in variables from config file
+	cores = config_dict['ECFLOW']['Cores']
+	run_dir = config_dict['VIC']['RunDir']
+	log_dir_root = config_dict['VIC']['LogDirRoot']
+	forcing_names = config_dict['COMBINE']['Forcing_Name_List']
+	script = config_dict['COMBINE']['Script'] #location of original tocel_combine.py 
+	tmin_dir = config_dict['COMBINE']['Tmin_dir']
+	tmax_dir = config_dict['COMBINE']['Tmax_dir']
+	precip_dir = config_dict['COMBINE']['Precip_dir']
+	wind_dir = config_dict['COMBINE']['Wind_dir']
+	final_dir = config_dict['COMBINE']['Final_Met_dir']
+	
+	#set up multiprocessor
+	if cores > 1:
+        	pool = mp.Pool(processes=cores)
+        	logger.debug('started pool with {} processes'.format(cores))
+
+
+
+	#read in list of lat/lons
+	lat_lon = np.genfromtxt('%s' %(forcing_names), dtype='str')
+
+	lat_lons=[]
+	lat_lons_list=[]
+	log_dir_list = []
+
+	#in run_dir make a tocel_combine script for every grid cell 
+	for ll in lat_lon:
+		log_dir = '{0}{1}'.format(log_dir_root, ll)
+		lat_lons = os.path.join(
+                	run_dir, 'control_{0}'.format(ll))
+		copyfile(script, lat_lons)
+		replace(lat_lons, 'DATA_LAT_LON', ll)
+		replace(lat_lons, 'TMIN_DIREC', tmin_dir)
+		replace(lat_lons, 'TMAX_DIREC', tmax_dir)
+		replace(lat_lons, 'PRECIP_DIREC', precip_dir)
+		replace(lat_lons, 'WIND_DIREC', wind_dir)
+		replace(lat_lons, 'FINAL_DIREC', final_dir)
+		lat_lons_list.append(lat_lons)
+		log_dir_list.append(log_dir)
+
+	#run each tocel_combine script using subprocess
+	if cores > 1:
+	        try:
+	        	logger.info('about to apply_async')
+            		for i in lat_lons_list:
+                 		run_tocel_combine = ['python', '%s' %(i)]
+                 		pool.apply_async(proc_subprocess, args=(run_tocel_combine, run_dir))
+		except (mp.ProcessError, mp.TimeoutError, mp.BufferTooShort,
+        	         mp.AuthenticationError) as e:
+             		logger.info(e)
+
+	else:
+        	print('core set to 1')
```

Comment:
You can have this print using `logger.info('core set to 1')`, in which case it will give more information about which function the statement comes from.